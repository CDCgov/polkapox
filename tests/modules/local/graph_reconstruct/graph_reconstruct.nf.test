nextflow_process {

    name "Test Process GRAPH_RECON"
    script "modules/local/graph_reconstruct/graph_reconstruct.nf"
    process "GRAPH_RECON"

    test("Graph Recon Test") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test_sub', single_end: true ],
                    file('${projectDir}/tests/nf-test-data/test_sub.bridges_applied.gfa', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
