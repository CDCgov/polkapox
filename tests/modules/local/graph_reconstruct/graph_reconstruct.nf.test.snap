{
    "Graph Recon Test": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "gfa_assembly": [
                    
                ],
                "log": [
                    
                ],
                "summary": [
                    
                ],
                "unicycler_contigs": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-07T16:42:49.537674556"
    }
}